client/web/src/components/RedirectRoute.tsx

```diff
@@ -0,0 +1,16 @@
+import { FC } from 'react'
+
+import { Location, Navigate, Params, useLocation, useParams } from 'react-router-dom-v5-compat'
+
+interface RedirectRouteProps {
+    getRedirectURL: (data: { location: Location; params: Params }) => string
+}
+
+export const RedirectRoute: FC<RedirectRouteProps> = props => {
+    const { getRedirectURL } = props
+
+    const location = useLocation()
+    const params = useParams()
+
+    return <Navigate to={getRedirectURL({ location, params })} />
```

Comment:
Redirects should replace, no? Otherwise the back button will just redirect again :D

```suggestion
    return <Navigate replace={true} to={getRedirectURL({ location, params })} />
```